ONSdigital/census-rh-ui
# CR-730 Address in Scotland

MUST BE DEPLOYED WITH/AFTER MATCHING CUCUMBER UPDATE AND INGRESS UPDATE

# Motivation and Context
This change takes advantage of the update to AIMS to include the new 'countryCode', enabling detection of addresses that are not in England, Wales or Northern Ireland.
Resolves CR-730 and CR-776

# What has changed
- New step in ConfirmAddress when 'yes' is submitted to do a UPRN query against AIMS. This returns the countryCode, checks if value is 'S' and if so displays a page telling the user to go to the Scottish Census site. If not 'S' (or no value) carries on as normal.
- Change 'NotRequired' error when uprn is not found in case query to 'Contact CCCC' (same logic but different page)
- Some naming tweaks to shift to Request from RequestCode, to make things more generic for further fulfilment journeys to be added
- Change to fulfilments url to start {display_region}/requests/ to simplify ingress and overall urls

# How to test?
Tested on dev with specific cucumber updates (see links)
Scottish part can only be tested in dev currently, as new AIMS fields are not available in the off-network version. If countryCode is not present in AIMS response, should see a single log line and carry on past the code.

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-730
https://collaborate2.ons.gov.uk/jira/browse/CR-776
https://github.com/ONSdigital/census-rh-cucumber/pull/53

# Screenshots (if appropriate):
